### PR TITLE
[FIX] im_livechat: fix chatbot edge cases

### PR DESCRIPTION
**Pull request by awa-odoo**
(Apply rebase-ff)

### CODE REVIEW

File: addons/im_livechat/models/chatbot_script_step.py [hunk 1]
```diff
@@ -334,7 +334,9 @@ def _process_step_forward_operator(self, mail_channel):
         if mail_channel.livechat_channel_id:
             human_operator = mail_channel.livechat_channel_id._get_random_operator()
 
-        if human_operator:
+        # handle edge case where we found yourself as available operator -> don't do anything
+        # it will act as if no-one is available (which is fine)
+        if human_operator and human_operator != self.env.user:
```
Comment (tde-banana-odoo) [inline addons/im_livechat/models/chatbot_script_step.py:339]: Shouldn't we exclude ourself from get_random_operator in that case ?

File: addons/im_livechat/static/src/legacy/public_livechat_chatbot.js [hunk 2]
```diff
@@ -432,6 +440,10 @@ const QWeb = core.qweb;
      * @private
      */
     _chatbotShouldEndScript: function () {
+        if (this._chatbotCurrentStep['conversation_closed']) {
```
Comment (tde-banana-odoo) [inline addons/im_livechat/static/src/legacy/public_livechat_chatbot.js:443]: Why a dict-like operation by the way ?

**Review by tde-banana-odoo — COMMENTED** [review]
Much review, amazing reading

File: addons/im_livechat/static/src/legacy/public_livechat_chatbot.js (hunk 2 above)
Comment (awa-odoo) [reply addons/im_livechat/static/src/legacy/public_livechat_chatbot.js]: Oopsie :disguised_face:

File: addons/im_livechat/models/chatbot_script_step.py (hunk 1 above)
Comment (awa-odoo) [reply addons/im_livechat/models/chatbot_script_step.py:339]: That would technically be the "good" fix to do.

Here I went for the "cheap" one as it requires changing the API.
(Or calling _get_random_operator multiple times but that's really bad design).

And honestly I don't really feel like this is worth it considering this is a very edge case, I'm ok with it just "not crashing the script".

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**